great_expectations/metrics/metric_results.py

```diff
@@ -23,20 +22,14 @@ class Config:
         arbitrary_types_allowed = True
 
 
-class MetricErrorResultValue(TypedDict):
-    exception_info: ExceptionInfo
+class MetricErrorResultValue(BaseModel):
```

Comment:
The MetricErrorResultValue isn't public or returned by anything public so it is ok to change. If you wanted to make it backwards compatible you could do 
```
class MetricErrorResultValue(BaseModel):
  exception_info: ExceptionInfo
  exception_message: str
  exception_traceback: Union[str, None] = None
```

Reply:
This is returned by the API. This is what you can find in `MetricErrorResult.value`. We had it the way you show in the first incarnation of `MetricErrorResult`, but its shape is confusing. `ExpectionInfo` includes duplicate `exception_message` and `exception_traceback` 😮‍💨